### PR TITLE
Fixed: Undo functionality in diagram #9915

### LINKED ISSUE #9915: Undo functionality in diagram
We'll want the user to be able to go back a step when pressing ctrl + z. I'm not sure how many steps back we should save but we can say about 15 for now to not overdo it. We'll want the old state to be saved when:

  - An element is moved
  - An element is created

In the future we'll want to include when an edge is created and possibly more, but since edge creation isn't fixed yet that'll have to be a future issue.

### PR DESCRIPTION
**Pull request by a19hammi**
Pretty much all forms of undoing works.
Since we're missing the storage of PREVIOUS values during a value change there's no way to RESTORE values when stepBack is called.

In other words. We can now step back on all kinds of changes EXCEPT attributes.

fixes #9915

### CODE REVIEW

**Comment by a19hammi** [comment]
Reminder for **_reviewers_**. Attribute step-back will be handled in a future issue later since we rely on #10240 to be solved first.

**Review by a19chrel — CHANGES_REQUESTED** [review]
The undo-functionallity does not support removal of created lines while stepping back in the history. Also your if-statements and for-loop does not follow the code standard.

File: DuggaSys/diagram.js [hunk 1]
```diff
@@ -127,20 +142,31 @@ class StateChange
...
             this.name = changes.name;
         }
+
         if (changes.moved) {
-            if (this.moved) this.moved.add(changes.moved);
-            else this.moved = changes;
+            if (this.moved) { 
+                this.moved.add(changes.moved);
+            }
+            else {
```
Comment (a19chrel) [inline DuggaSys/diagram.js:150]: Both brackets should be on the same row.

**Review by a19chrel — CHANGES_REQUESTED** [review]
Some small changes are needed to meet the code standard.

**Review by a19chrel — APPROVED** [review]
LGTM!

**Pull request merged**